marklogic-client-api/src/main/java/com/marklogic/client/document/DocumentWriteOperation.java

```diff
@@ -85,4 +90,88 @@
    * @return the logical temporal document URI
    */
   String getTemporalDocumentURI();
+  
+  /**
+   * The from method prepares each content object for writing as a document including generating a URI by inserting a UUID.
+   * @param a subclass of AbstractWriteHandle
+   * @param DocumentUriMaker which internally accepts an AbstractWriteHandle and returns a String
+   * @return a stream of DocumentWriteOperation to be written in the database.
+   */
+    public static Stream<DocumentWriteOperation> from(Stream<? extends AbstractWriteHandle> content,
+            final DocumentUriMaker uriMaker) {
+        final class DocumentWriteOperationImpl implements DocumentWriteOperation {
+            public DocumentWriteOperationImpl(AbstractWriteHandle content, String uri) {
+                this.content = content;
+                this.uri = uri;
+            }
+
+            private AbstractWriteHandle content;
+            private String uri;
```

Comment:
For consistency and readability, should the fields move to the top of the class (that is, before the constructor)?